cmd/agent/gui/platform_darwin.go

```diff
@@ -8,14 +8,18 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/config"
 )
 
+func restartEnabled() bool {
+	return false
+}
+
 func restart() error {
 	return fmt.Errorf("restarting the agent is not implemented on non-windows platforms")
 }
 
 // writes auth token(s) to a file with the same permissions as datadog.yaml
-func saveAuthToken(token string) error {
+func saveAuthToken() error {
```

Comment:
`saveAuthToken` relies on the `authTokenPath` global var, meaning that if we call it before `fetchAuthToken()` it would fail. It's a nit but I'd pass the path to the token in a param, making it obvious what the function needs to work properly